lib/chef/knife/vsphere_vm_vnc_set.rb

```diff
@@ -0,0 +1,50 @@
+require 'chef/knife'
+require 'chef/knife/base_vsphere_command'
+require 'rbvmomi'
+require 'netaddr'
+
+class Chef::Knife::VsphereVmVncset < Chef::Knife::BaseVsphereCommand
+  banner 'knife vsphere vm vncset VMNAME VNCPORT VNCPASSWORD'
+
+  common_options
+
+  def run
+    $stdout.sync = true
+    vmname = @name_args[0]
+    if vmname.nil?
+      show_usage
+      fatal_exit('You must specify a virtual machine name')
+    end
+
+    vnc_port = @name_args[1]
```

Comment:
What would you think of making the port and password an option rather than a positional command line parameter?